service-workers/service-worker/installing.https.html

```diff
@@ -5,33 +5,38 @@
 <script src="resources/test-helpers.sub.js"></script>
 <body>
 <script>
-// "installing" is set
-async_test(function(t) {
-    var step = t.step_func.bind(t);
-    var url = 'resources/empty-worker.js';
-    var scope = 'resources/blank.html';
-    var frame;
 
-    service_worker_unregister(t, scope)
-    .then(step(function() { return with_iframe(scope); }))
-    .then(step(function(f) {
-        frame = f;
-        return navigator.serviceWorker.register(url, {scope: scope});
-    }))
-    .then(step(function(registration) {
-        var container = frame.contentWindow.navigator.serviceWorker;
-        assert_equals(container.controller, null);
-        assert_equals(registration.active, null);
-        assert_equals(registration.waiting, null);
-        assert_equals(registration.installing.scriptURL, normalizeURL(url));
+const SCRIPT = 'resources/empty-worker.js';
+const SCOPE = 'resources/blank.html';
+
+// "installing" is set
+promise_test(async t => {
+  await service_worker_unregister(t, SCOPE);
+  const frame = await with_iframe(SCOPE);
+  const registration =
+      await navigator.serviceWorker.register(SCRIPT, {scope: SCOPE});
+  const container = frame.contentWindow.navigator.serviceWorker;
+  assert_equals(container.controller, null, 'controller');
+  assert_equals(registration.active, null, 'registration.active');
+  assert_equals(registration.waiting, null, 'registration.active');
+  assert_equals(registration.installing.scriptURL, normalizeURL(SCRIPT),
+                'registration.active');
+  // FIXME: Add a test for a frame created after installation.
+  // Should the existing frame ("frame") block activation?
 
-        // FIXME: Add a test for a frame created after installation.
-        // Should the existing frame ("frame") block activation?
-    }))
-    .then(step(function() {
-        frame.remove();
-        return service_worker_unregister_and_done(t, scope);
-    }))
-    .catch(unreached_rejection(t));
+  frame.remove();
+  await service_worker_unregister_and_done(t, SCOPE);
 }, 'installing is set');
+
+// Tests that The ServiceWorker objects returned from installing attribute getter
+// that represent the same service worker are the same objects.
+promise_test(async t => {
+  const registration =
+      await service_worker_unregister_and_register(t, SCRIPT, SCOPE);
+  const r = await navigator.serviceWorker.getRegistration(SCOPE);
+  assert_equals(registration.installing, r.installing, 'getRegistration ' +
```

Comment:
Btw, I think the spec requires registration1 == registration2, which makes this test not so valuable. But we can keep it for now.

Reply:
Yes, but in this case, the spec requires the `installing` attribute should return `ServiceWorker` object instead of `ServiceWokerRegistration` object, how about using `registration1.installing == registration2.installing`?